### PR TITLE
Fixes #54: Add cli completion

### LINKED ISSUE #54: [Feature request] Add cli completion
I would be a nice expierence to tab trough the cli options. 

## Expected Behaviour

Tab trough the cli options

## Current Behaviour

No tabbing

## Possible Solution

https://github.com/spf13/cobra/blob/master/bash_completions.md

### PR DESCRIPTION
**Pull request by pableeee**
Fixes #54

Enables completion for the cli options

## Description
This change enables cli completion feature, for both bash and zsh, by the use of the tab key (given all requirement are met, see  https://docs.openfaas.com/cli/completion)

## Motivation and Context
Enhance cli user experience by hinting cli options and completing them once they are uniquely idenfied. 

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Tested manually in both bash and zsh shells.
```
$ arkade help completion
Generates shell auto completion for Bash or ZSH.

Please follow the instructions in the link below to activate the shell auto completion in your environment:
https://docs.openfaas.com/cli/completion/

Usage:
  arkade completion SHELL [flags]

Flags:
  -h, --help           help for completion
      --shell string   Outputs shell completion, must be bash or zsh
$ . <(arkade completion --shell bash)

$ arkade in
info     install

$ arkade install
cert-manager             crossplane               grafana                  istio                    linkerd                  mongodb                  openfaas-ingress
chart                    docker-registry          info                     kafka-connector          metrics-server           nginx-ingress            postgresql
cron-connector           docker-registry-ingress  inlets-operator          kubernetes-dashboard     minio                    openfaas                 traefik2

$ arkade install docker-registry
docker-registry          docker-registry-ingress

$ arkade install docker-registry-ingress
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/completion.go [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+var shell string
+
+func MakeCompletion() *cobra.Command {
+	var completionCmd = &cobra.Command{
+		Use:   "completion SHELL",
+		Short: "Generates shell completion",
+		Long: `Generates shell auto completion for Bash or ZSH.
+		
+Please follow the instructions in the link below to activate the shell auto completion in your environment:
+https://docs.openfaas.com/cli/completion/`,
```
Comment (aidun) [inline cmd/completion.go]: I think this link is not correct.

File: cmd/completion.go [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	}
+
+	completionCmd.Flags().StringVar(&shell, "shell", "", "Outputs shell completion, must be bash or zsh")
+	completionCmd.MarkFlagRequired("shell")
+
+	return completionCmd
+}
+
+func runCompletion(cmd *cobra.Command, args []string) (err error) {
+	if shell == "" {
```
Comment (aidun) [inline cmd/completion.go]: This can be done with the default case of the switch statement.

File: main.go [hunk 3]
```diff
@@ -1,6 +1,3 @@
-// Copyright (c) arkade author(s) 2020. All rights reserved.
```
Comment (aidun) [inline main.go]: I think this shouldn´t be removed.

File: cmd/completion.go (hunk 1 above)
Comment (pableeee) [reply cmd/completion.go]: I took this snippet from faas-cli project, so users could have a little bit of a context on how to set up the completion.
I could include these instructions on the link to the readme.md, and remove this from the completion command help.
Sounds good?

File: cmd/completion.go (hunk 2 above)
Comment (pableeee) [reply cmd/completion.go]: kept this validation, but its now used to read from the environment when shell argument is missing.

File: main.go (hunk 3 above)
Comment (pableeee) [reply main.go]: 100% right 😋. Fixed

**Comment by alexellis** [comment]
Hi, can you please amend your commit message so that it doesn't wrap past 72 characters?

See the screenshot for what I mean:

Screenshot 2020-04-03 at 09 20 56

Whilst you are making the change, please also squash the three commits into one:

```
git rebase -i HEAD~3
```

File: cmd/completion.go [hunk 4]
```diff
@@ -0,0 +1,233 @@
+package cmd
```
Comment (alexellis) [inline cmd/completion.go:1]: Add the project copyright notice please.

Comment (pableeee) [reply cmd/completion.go:4]: no problem

**Comment by pableeee** [comment]
commits squashed and copyright notice added
Cheers :)

**Comment by pableeee** [comment]
@alexellis @aidun 
Want to give it another look?

**Comment by aidun** [comment]
I will have a look at it and will provide feedback tomorrow.

**Comment by alexellis** [comment]
@aidun are you still happy to take a look at this, the last message was a couple of months ago now?

@pableeee thanks for the PR - I see you're waiting for feedback from @aidun before I look at merging this. My initial suggestion would be that you rebase since we can't merge at the moment due to a couple of conflicts displayed above.

Thanks,

Alex

**Comment by aidun** [comment]
@pableeee sorry for the delay. I will test and review it after the changes suggests by Alex.

**Comment by pableeee** [comment]
@alexellis @aidun, No worries.

I rebased my branch to master, so we should be able to merge it if everything looks good.

Regards,
Pable

**Comment by aidun** [comment]
@pableeee on Mac with zsh it works as expected, but not for all sub commands, think it is how the subcommands are implemented, I have to go trough the code...

Example: 
```
arkade install <tab> # -> list of commands
arkade info <tab> # nothing
```

**Comment by pableeee** [comment]
@aidun doing a little [refactor](https://github.com/pableeee/arkade/pull/1) on the info command, we can get those completions as well.

**Comment by alexellis** [comment]
@pableeee can you compare to #159 and see if we missed anything there?

**Pull request closed without merging**